pmm/v3/pmm3-package-tests-matrix.groovy

```diff
@@ -2,17 +2,121 @@ library changelog: false, identifier: 'lib@master', retriever: modernSCM([
+        string(name: 'CLIENTS', value: CLIENTS),
+        string(name: 'NOTIFY', value: 'false'),
+        string(name: 'DAYS', value: '1')
+    ]
+    env.VM_IP = stagingJob.buildVariables.IP
+    env.PMM_SERVER_IP = stagingJob.buildVariables.IP
+    env.VM_NAME = stagingJob.buildVariables.VM_NAME
+    env.ADMIN_PASSWORD = stagingJob.buildVariables.ADMIN_PASSWORD
+    env.PMM_URL = "http://admin:${ADMIN_PASSWORD}@${VM_IP}"
+}
+
+void runPackageTest(String GIT_BRANCH, DOCKER_VERSION, PMM_VERSION, TESTS, INSTALL_REPO, TARBALL, METRICS_MODE, CLIENTS) {
+    packageTestJob = build job: 'pmm3-package-testing-arm', parameters: [
         string(name: 'GIT_BRANCH', value: GIT_BRANCH),
         string(name: 'DOCKER_VERSION', value: DOCKER_VERSION),
         string(name: 'PMM_VERSION', value: PMM_VERSION),
         string(name: 'TESTS', value: TESTS),
         string(name: 'INSTALL_REPO', value: INSTALL_REPO),
-        string(name: 'METRICS_MODE', value: METRICS_MODE)
+        string(name: 'TARBALL', value: TARBALL),
+        string(name: 'METRICS_MODE', value: METRICS_MODE),
+        string(name: 'CLIENTS', value: CLIENTS)
     ]
 }
 
+def generateVariants(String playbookName) {
+    def results = new HashMap<>();
+    def labels = ["min-bookworm-arm64", "min-bullseye-arm64", "min-noble-arm64", "min-jammy-arm64", "min-focal-arm64", "min-ol-9-arm64", "min-ol-8-arm64"]
+
+
+    for(label in labels) {
+        results.put("${label}-${playbookName}", generateStage(label, playbookName))
+    }
+
+    return results;
+}
+
+def generateRunnerVariants() {
+    def results = new HashMap<>();
+    def agents = ["min-bookworm-arm64", "min-bullseye-arm64"]
+    def playbooks = ["pmm3-client", "pmm3-client_custom_path", "pmm3-client_integration", "pmm3-client_integration_auth_config", "pmm3-client_integration_auth_register", "pmm3-client_integration_custom_path", "pmm3-client_integration_custom_port", "pmm3-client_integration_upgrade", "pmm3-client_integration_upgrade_custom_path", "pmm3-client_integration_upgrade_custom_port", "pmm3-client_upgrade"]
+
+    for(agent in agents) {
+        for(playbook in playbooks) {
+            results.put("${agent}-${playbook}", generateStage(agent, playbook))
+        }
+    }
+
+    return results;
+}
+
+
+def generateStage(LABEL, PLAYBOOK) {
+    return {
+        stage("${LABEL}-${PLAYBOOK}") {
+            retry(5) {
+                agent {
+                    label "${LABEL}"
+                }
+                node(LABEL) {
+                    String DISTRIBUTION = sh(script: "cat /proc/version", , returnStdout: true).trim()
+                    if(DISTRIBUTION.contains("Red Hat")) {
+                        sh '''
+                            sudo yum install -y epel-release
+                            sudo yum -y update
+                            sudo yum install -y ansible-core git wget dpkg
```

Comment:
```suggestion
                            sudo dnf install -y epel-release
                            sudo dnf -y update
                            sudo dnf install -y ansible-core git wget dpkg
```

Reply:
I'm suggesting this because `yum` is essentially a symlink to `dnf`.

Reply:
Changes done.